packages/lesswrong/server/repos/PostsRepo.ts

```diff
@@ -347,4 +348,25 @@ export default class PostsRepo extends AbstractRepo<DbPost> {
     const {count} = await this.getRawDb().one(`SELECT COUNT(*) FROM "Posts"`);
     return count;
   }
+
+  getDialogueResponseCount = (post:DbPost) => {
+    const html = post.contents.originalContents.data;
+    const parsedHtml= cheerioParse(html);
+    
+    const blocksWithId = parsedHtml('block[message-id]');
+    const ids : string[] = blocksWithId.map( (i, block) => parsedHtml(block).attr('message-id')).get();
+    
+    return ids.length;
+  }
+
+  getDialogueMessageTimestamps = (post: DbPost): Date[] => {
+    const html = post.contents.originalContents.data;
+    const parsedHtml= cheerioParse(html);
+  
+    const blocksWithId = parsedHtml('block[message-id]');
+    const timestampStrings = blocksWithId.map( (i, block) => (parsedHtml(block).attr('submitted-at'))).get();
+    const timestamps = timestampStrings.map( dateString => new Date(dateString))
+    
+    return timestamps
+  }
```

Comment:
I think these shouldn't be in `PostsRepo` - they're used for SQL queries on the respective table.  If we don't have an appropriate place to put these, I'd create some new server-side utils file for these.